### PR TITLE
Add spoiler template action button

### LINKED ISSUE #1190: Add spoiler template action button
### Pre-Flight checklist

- [X] Did you check to see if this issue already exists?
- [X] This is a single feature request. (Do not put multiple feature requests in one issue)
- [X] This is not a question or discussion. (Use https://lemmy.ml/c/jerboa for that)

### Describe The Feature Request Below

Add spoiler template action button, like we have bold, italics

### PR DESCRIPTION
**Pull request by MV-GH**
- Add spoiler template action button
- Refactored the markdownhelperbar: so that the actions are inside the composable instead of being passed 
- Changed spoiler plugin itself to add a newline, so that the first line of the spoiler content does not get put after the title but instead on a new line
- The included icon is from [material3 symbols](https://fonts.google.com/icons?icon.query=emergency+home&icon.style=Outlined), not yet available as dep for Android 

https://github.com/dessalines/jerboa/assets/67873169/b84e4a4a-6890-4f59-ab23-d272b1c02711

Fixes #1190

### CODE REVIEW

File: app/src/main/res/drawable/emergency_home_fill0_wght400_grad0_opsz48.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (dessalines) [inline app/src/main/res/drawable/emergency_home_fill0_wght400_grad0_opsz48.xml:1]: The icon in lemmy-ui is 

Screenshot_2023-08-29-10-52-38-142_mark via gp

Which should be available as https://fonts.google.com/icons?selected=Material+Icons+Outlined:report_problem:&icon.set=Material+Icons&icon.query=warning or Warning Amber.

I don't know if there's a universal spoiler icon that ppl have settled on, but it should probably match lemmy-ui at least for now.

Comment (dessalines) [reply app/src/main/res/drawable/emergency_home_fill0_wght400_grad0_opsz48.xml:1]: Everything else looks good.

Comment (MV-GH) [reply app/src/main/res/drawable/emergency_home_fill0_wght400_grad0_opsz48.xml:1]: The icon I used was the one Reddit and other client used for Spoilers

Comment (dessalines) [reply app/src/main/res/drawable/emergency_home_fill0_wght400_grad0_opsz48.xml:1]: mmk then, I'll make an issue for lemmy-ui to add this one.

Comment (MV-GH) [reply app/src/main/res/drawable/emergency_home_fill0_wght400_grad0_opsz48.xml:1]: Well I am not against changing icon, especially if Lemmy also supports that kind of spoiler that the icon was used for

**Pull request merged**